### PR TITLE
Add libopusenc recipe

### PR DESCRIPTION
**Pull request by asafkahlon**
Library for encoding .opus audio files and live streams.

Signed-off-by: Asaf Kahlon <asafka7@gmail.com>

Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by asafkahlon** [comment]
@conda-forge/help-c-cpp is there anything else I should do in this PR?

File: recipes/libopusenc/build.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+./configure --prefix=${PREFIX} --enable-static --enable-shared
```
Comment (scopatz) [inline recipes/libopusenc/build.sh]: Please remove enable-static.  We prefer not to ship static binaries, and prefer to link instead. Thanks!

File: recipes/libopusenc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  sha256: 8298db61a8d3d63e41c1a80705baa8ce9ff3f50452ea7ec1c19a564fe106cbb9
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - pkgconfig
```
Comment (scopatz) [inline recipes/libopusenc/meta.yaml]: Please add `make` as a build dep

File: recipes/libopusenc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  host:
+    - libopus
+  run:
+    - libopus
+
+test:
+  commands:
+    - test -f ${PREFIX}/include/opus/opusenc.h
+    - test -f ${PREFIX}/lib/libopusenc.a
```
Comment (scopatz) [inline recipes/libopusenc/meta.yaml]: Please test the share shared lib  you can use `test -f ${PREFIX}/lib/libopusenc${SHLIB_EXT}` IIRC

**Comment by scopatz** [comment]
A few small edits, thanks!

**Comment by asafkahlon** [comment]
Thanks for the review :)
I pushed another revision.

**Pull request merged**